src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -235,7 +242,19 @@ private void initSearchModifierButtons() {
         fulltextButton.setTooltip(new Tooltip(Localization.lang("Fulltext search")));
         initSearchModifierButton(fulltextButton);
         fulltextButton.selectedProperty().addListener((obs, oldVal, newVal) -> {
-            searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, newVal);
+            if (!filePreferences.shouldFulltextIndexLinkedFiles() && newVal) {
+                boolean enableFulltextSearch = dialogService.showConfirmationDialogAndWait(Localization.lang("Fulltext search"), Localization.lang("Fulltext search requires the setting 'Automatically index all linked files for fulltext search' to be enabled. Do you want to enable indexing now?"), Localization.lang("Enable indexing"), Localization.lang("Keep disabled"));
+
+                LibraryTab libraryTab = tabContainer.getCurrentLibraryTab();
+                if (libraryTab != null && enableFulltextSearch) {
+                    filePreferences.setFulltextIndexLinkedFiles(true);
+                    libraryTab.getLuceneManager().rebuildIndex();
+                }
+                if (!enableFulltextSearch) {
+                    fulltextButton.setSelected(false);
+                }
+            }
+            searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, true);
```

Comment:
Why is the flag always set to true?

Reply:
I think the flag and the toggle button should be true/selected only if indexing is enabled.

Reply:
Oh that was some leftover from my previous experiments. Thanks for spotting!